### PR TITLE
Let Travis.config.prefer_https override git URL scheme

### PR DESCRIPTION
**Pull request by BanzaiMan**
This PR introduces config parameter `prefer_https`, which forces the repo's source URL to HTTPS.

### CODE REVIEW

**Comment by BanzaiMan** [comment]
This is a replacement for https://github.com/travis-ci/travis-scheduler/pull/30.

File: lib/travis/scheduler/serialize/worker/repo.rb [hunk 1]
```diff
@@ -25,7 +25,7 @@ def api_url
           end
 
           def source_url
-            private? || force_private? ? source_git_url : source_http_url
+            ((private? || force_private?) && !!!Travis.config.prefer_https) ? source_git_url : source_http_url
```
Comment (svenfuchs) [inline lib/travis/scheduler/serialize/worker/repo.rb]: `!!!Travis.config.prefer_https`

not not not?

File: lib/travis/scheduler/serialize/worker.rb [hunk 2]
```diff
@@ -114,6 +115,13 @@ def cache_config
           def format_date(date)
             date && date.strftime('%Y-%m-%dT%H:%M:%SZ')
           end
+
+          def github_oauth_token
+            candidates = job.repository.users.where("github_oauth_token IS NOT NULL").
+                    order("updated_at DESC")
+            admin = candidates.first
+            admin && admin.github_oauth_token
```
Comment (svenfuchs) [inline lib/travis/scheduler/serialize/worker.rb]: This user may or may not have a valid token here, depending on the sync status. Other places solve this by trying various tokens. I'm not sure what the best advice is though. I guess we don't have a way of making sure an admin token is valid atm.

File: lib/travis/scheduler/serialize/worker/repo.rb (hunk 1 above)
Comment (BanzaiMan) [inline lib/travis/scheduler/serialize/worker/repo.rb]: I suppose I got carried away. I can dial down to just one `!`.

File: lib/travis/scheduler/serialize/worker.rb (hunk 2 above)
Comment (BanzaiMan) [inline lib/travis/scheduler/serialize/worker.rb]: There is always a chance that this user's key is invalid. The intention is that if the user was sync'd most recently, their key is least likely to be. This, of course, is questionable assumption.

We could send multiple keys in the payload for `travis-build` to try, but that will be more involved.

**Comment by svenfuchs** [comment]
Looks good to me, @BanzaiMan!

**Pull request merged**